stan/math/rev/fun/from_var_value.hpp

```diff
@@ -25,6 +25,45 @@ Eigen::Matrix<var, T::RowsAtCompileTime, T::ColsAtCompileTime> from_var_value(
   return res;
 }
 
+/**
+ * Forward Eigen containers of vars.
+ *
+ * @tparam T type of the input
+ * @param a matrix to convert
+ */
+template <typename T, require_eigen_vt<is_var, T>* = nullptr>
+auto from_var_value(T&& a) {
+  return std::forward<T>(a);
+}
+
+/**
+ * Forward scalar vars on.
+ *
+ * @tparam T type of the input
+ * @param a matrix to convert
+ */
+template <typename T, require_var_vt<std::is_arithmetic, T>* = nullptr>
+auto from_var_value(T&& a) {
+  return std::forward<T>(a);
+}
+
+/**
+ * Convert the elements of the `std::vector` input to `var_value` types
+ * if possible
+ *
+ * @tparam T type of elemnts of the input vector
+ * @param a std::vector of elements to convert
+ */
+template <typename T>
+auto from_var_value(const std::vector<T>& a) {
+  std::vector<decltype(from_var_value(std::declval<T>()))> out;
```

Comment:
Should `decay` this `decltype`? It would definitely be okay to decay it, but I'm not sure it's necessary in any reasonable case.

Reply:
I think it is not necessary